docs/book/src/faq.md

```diff
@@ -0,0 +1,88 @@
+For example, if set `--domain=example.com`:
+```
+kubebuilder init --domain example.com --repo xxx --plugins=go/v4-alpha
+kubebuilder create api --group mygroup --version v1beta1 --kind Mykind
+```
+Then the result resource group will be `mygroup.example.com`.
+
+If domain field not set, the default value is `my.domain`.
+
+## How to validate API or set default value?
+We can use `webhook` to achieve it. See [webhook](./reference/webhook-overview.md)
+
+For example, we can create webhook with defaulting and validating logic by:
+```
+kubebuilder create webhook --group mygroup --version v1beta1 --kind Mykind --defaulting --programmatic-validation
+```
+Then the generated webhook code is like this:
+```go
+func (r *Mykind) Default() {
+	// TODO(user): fill in your defaulting logic.
+}
+
+func (r *Mykind) ValidateCreate() error {
+	// TODO(user): fill in your validation logic upon object creation.
+	return nil
+}
+
+func (r *Mykind) ValidateUpdate(old runtime.Object) error {
+	// TODO(user): fill in your validation logic upon object update.
+	return nil
+}
+
+func (r *Mykind) ValidateDelete() error {
+	// TODO(user): fill in your validation logic upon object deletion.
+	return nil
+}
+```
+So it is easy to add API default and validate logic.
+
+## How to use `klog` or other loggers as the project logger?
+We can set the project logger by:
+```go
+	//ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+	ctrl.SetLogger(klog.NewKlogr())
+```
+
+## Testing leader-election outside of the cluster?
+If you are testing the project locally using the `make run` target which will run the manager outside of the cluster, then you might also need to set the namespace, as follows:
+```go
+mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme:                 scheme,
+		MetricsBindAddress:     metricsAddr,
+		Port:                   9443,
+		HealthProbeBindAddress: probeAddr,
+		LeaderElection:         enableLeaderElection,
+		LeaderElectionID:       "14be1926.testproject.org",
+		LeaderElectionNamespace: "your-ns", // set this field to `make run` locally
+```
+
+## Got controller-pod SA permission denied error?
+Based on the quick start doc, did the make manifests, make install, then
+make docker-build docker-push IMG=xxx/xxx:tag, make deploy IMG=xxx/xxx:tag, then
+in the cluster the controller-pod got error:
+```
+1.6656687258729894e+09  ERROR   controller-runtime.client.config        unable to get kubeconfig        {"error": "open /var/run/secrets/kubernetes.io/serviceaccount/token: permission denied"}
+sigs.k8s.io/controller-runtime/pkg/client/config.GetConfigOrDie
+        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.0/pkg/client/config/config.go:153
+main.main
+        /workspace/main.go:68
+runtime.main
+        /usr/local/go/src/runtime/proc.go:250
+```
+
+Reason: see [SA token permission related issues and PRs](https://github.com/kubernetes-sigs/kubebuilder/issues/3028#issuecomment-1288114462)
```

Comment:
It does not occur from k8s version X, right? (ihmo)We need to better clarify it here, i.e.: 

If you are facing the error `{"error": "open /var/run/secrets/kubernetes.io/serviceaccount/token: permission denied"}` when you are running the project against a Kubernetes cluster < X it might be caused by the issue (add links) so, the reason is ..., see ... here. Then, the workaround is: 


However, not that this problem is fixed and will not occurs if you deploy the project in versions >= X.